woocommerce/woocommerce-ios
# [Migrate simple payments] Navigate to payment methods after creating an order & collecting payment

Part of #12387 
⚠️ Please make sure https://github.com/woocommerce/woocommerce-ios/pull/12400 is approved first if it hasn't been merged


## Why

As we're migrating from simple payments to order creation, we want to show the order form with a bottom sheet instead of the simple payments when the user taps on Payments > Collect Payment. Order creation has two completion callbacks, `onFinished` and `onFinishAndCollectPayment`. `onFinished` was implemented in https://github.com/woocommerce/woocommerce-ios/pull/12400, but as mentioned there `onFinishAndCollectPayment` requires more changes to navigate to the payment methods view. As a result, this PR implements the payment collection flow in order creation.

## How

Because [`navigationDestination(item:destination:)`](https://developer.apple.com/documentation/swiftui/view/navigationdestination(item:destination:)) for a Hashable item is supported in iOS 17+, we still need a boolean binding to navigate to the payment methods screen using [`navigationdestination(ispresented:destination:)`](https://developer.apple.com/documentation/swiftui/view/navigationdestination(ispresented:destination:)). `presentPaymentMethods` is the boolean state that controls when the payment methods screen is shown, and `paymentMethodsViewModel` is set at the same time for `PaymentMethodsView`. Another tricky detail that I found in the simple payments UI test is the order created notice snackbar, and I decided to recover this UI/UX by exposing `notices` observable in `PaymentMethodsViewModel` and then observing & sending the notice in `InPersonPaymentsMenuViewModel`.

## Testing instructions


- Go to the menu tab
- Tap on Payments > Collect Payment --> the order creation form should be shown, and a sheet should appear shortly with info about migrating simple payments to order creation
- Tap `Add a custom amount` --> the migration sheet should be dismissed then the custom amount sheet should be shown
- Enter some custom amount
- Tap `Add Custom Amount`
- Tap `Collect Payment` at the bottom to collect payment --> it should navigate to the payment methods screen
- Tap `Cash`
- Tap `Mark Order as Complete` --> the view should be popped on completion, and a success notice should be shown in the payments menu

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1945542/4be93d13-9a17-4b6d-8c5b-57d7efbe7315


---
